### ISSUE TITLE
fail to build iOS in Xcode

### ISSUE THREAD

**Issue by LoTwT** [atilafassina/quantum#4]
## error report

An error occurred when using `pnpm tauri ios dev --open`.

It seems related to the custom build script `Build Rust Code` at the bottom of the `src-tauri/gen/apple/project.yml`.

I can't find the `libquantum_lib.a` at its target path `my-quantum/src-tauri/gen/apple`.

<img width="897" alt="image" src="https://github.com/atilafassina/quantum/assets/50254496/3cc0a74b-aec1-422d-9337-698d1637694d">

## repro steps

- degit atilafassina/quantum my-quantum
- pnpm install
- pnpm tauri ios init
- pnpm tauri ios dev --open

## env info

- System:
    - OS: macOS 14.3
    - CPU: (10) arm64 Apple M1 Pro
- Binaries:
    - Node: 20.10.0 
    - pnpm: 8.14.0

## additional context

It's my first time using Xcode and building iOS. So I'm unsure if this is a problem with this template or any other problem of myself.

**Comment by atilafassina** [comment]
Hello, @LoTwT !!
Thanks for such a detailed report.

This happens because XCode does not use your environment variables by default, so it can't find Node.

The command below should allow your build to pass this step.

```sh
defaults write com.apple.dt.Xcode UseSanitizedBuildSystemEnvironment -bool NO
```

Important to note that this is a system-wide config and that it won't persist once you reboot. Unfortunately, Apple does not provide any other way to solve this more comfortably for us...

**Comment by LoTwT** [comment]
Thank you for your help! ❤️ 
I tried the command above but it still didn't work.

Finally, I find the problem is BLAME to environment variables about Node.

I use `fnm` to manage my node versions and it works well itself but fails to work with Xcode. I try to set `export PATH="$(dirname $(which node)):$PATH"` manually and I can build successfully in Xcode. After, I removed what I set and it also works. 

I'm confused about what happened now. But at least, I can successfully start the ios dev server.

Thank you for your help and this awesome template again!
